### DISCUSSION TITLE
Sending data from HA to API in Smart-EVSe

### DISCUSSION THREAD

**Discussion by rvdgaag** [serkri/SmartEVSE-3#185] [Q&A]
Hi all,

i need some guidance how to send data from Home Assistant (L1/L2/L3 load) to the Smart-EVSI API.

I use the normal Smart-EVSE integration now, but also installed the MQTT version which also works fine.

The reason is I have a modbus 3 phase meter connected in my main switchboard via modbus to the smart-evse and all works fine, now I need the data from the modbus meter to be available in modbus-TCP for other devices so I want to push the data to the Smart EVSE via the API or MQTT.

Thanks in advance

**Comment by dingo35** [comment]
Why dont you keep your modbus-meter connected to your SmartEVSE, and then redistribute the data to the other devices by either MQTT (preferred) or modbus-tcp (experimental)?

**Comment by rvdgaag** [comment]
It is a wish to have for such data at least as possible points of failure.

I'm investing is a battery solution now and a Victron inverter, prefer to have all the data with as little as possible devices.

I like the smart-evse with you excellent firmware but I would like it more when it is connected to ethernet instead of wifi
